src/utils.ts

```diff
@@ -37,11 +38,17 @@ async function normalizeStore(source: string | zarr.Readable): Promise<zarr.Loca
       ]);
       store = ReferenceStore.fromSpec(json);
     } else {
-      const url = new URL(source);
-      // grab the path and then set the URL to the root
-      path = ensureAbosolutePath(url.pathname);
-      url.pathname = "/";
-      store = new zarr.FetchStore(url.href);
+      // try ZipFileStore first, fallback to FetchStore
+      try {
+        store = ZipFileStore.fromUrl(source);
+        await store.get("/.zgroup");
```

Comment:
Is this compatible with Zarr v3?

Reply:
true! I've replaced with using ZipFileStore's `.has("/")`